packages/terra-functional-testing/src/reporters/file-output-reporter/index.js

```diff
@@ -0,0 +1,206 @@
+   * @return null
+   */
+  fileNameCheck({ formFactor, locale, theme }, browserName, cid) {
+    const fileNameConf = ['fileOutput'];
+    if (locale) {
+      fileNameConf.push(locale);
+      this.resultJsonObject.locale = locale;
+    }
+
+    if (theme) {
+      fileNameConf.push(theme);
+      this.resultJsonObject.theme = theme;
+    }
+
+    if (formFactor) {
+      fileNameConf.push(formFactor);
+      this.resultJsonObject.formFactor = formFactor;
+    }
+
+    if (browserName) {
+      fileNameConf.push(browserName);
+    }
+
+    if (cid) {
+      fileNameConf.push(cid);
+    }
+
+    this.fileName = fileNameConf.join('-');
+  }
+
+  /**
+   * Set the package name to moduleName property if specsValue contains /package string
+   * @param {string} specsValue - File path of current spec file from runners
+   * @return null
+   */
+  setTestModule(spec) {
+    if (spec.indexOf(`${path.sep}packages${path.sep}`) > -1) {
+      return spec.split(`${path.sep}packages${path.sep}`).pop().split(path.sep)[0];
+    }
+    const packageJsonPath = path.join(process.cwd(), 'package.json');
+    if (fs.existsSync(packageJsonPath)) {
+      const packagejson = fs.readJsonSync(packageJsonPath);
+      return packagejson.name;
+    }
+    return process.cwd().split(path.sep).pop();
+  }
+
+  onRunnerEnd(runner) {
+    this.runners.push(runner);
+    this.printReport(runner);
+  }
+
+  getMessage(runner) {
+    const results = this.getResultDisplay();
+    if (results.length === 0) {
+      return null;
+    }
+    const testLinks = runner.isMultiremote
+      ? Object.entries(runner.capabilities).map(
+        ([instanceName, capabilities]) => this.getTestLink({
+          config: { ...runner.config, ...{ capabilities } },
+          sessionId: capabilities.sessionId,
+          isMultiremote: runner.isMultiremote,
+          instanceName,
+        }),
+      )
+      : this.getTestLink(runner);
+    const output = [
+      ...this.getHeaderDisplay(runner),
+      '',
+      ...results,
+      ...this.getFailureDisplay(),
+      ...(testLinks.length ? ['', ...testLinks] : []),
+    ];
+    const preface = `[${this.getEnviromentCombo(
+      runner.capabilities,
+      false,
+      runner.isMultiremote,
+    ).trim()} #${runner.cid}]`;
+    const divider = '------------------------------------------------------------------';
+    const prefacedOutput = output.map((value) => (value ? `${preface} ${value}` : preface));
+    return `${divider}\n${prefacedOutput}`;
+  }
+
+  printReport() {
+    const { runners } = this;
+    if (runners && runners.length) {
+      runners.forEach((runner, index) => {
+        // determine correct file name given configuration for run
+        if (index === 0) {
+          const { cid, capabilities, config } = runner;
+          const { browserName } = capabilities;
+          this.fileNameCheck(config.launcherOptions, browserName, cid);
+        }
+
+        this.moduleName = this.setTestModule(runner.specs[0]);
```

Comment:
We probably don't need [line 24](https://github.com/cerner/terra-toolkit/pull/665/files#diff-d8f1b66e8db21668273f6ad5ba417c4a5af04b1fcd0e46b5e2923ac1345fa743R24) and this.moduleName can just be a regular const instead of a class var.

Reply:
https://github.com/cerner/terra-toolkit/blob/8c50b42e3f8fdde85c9640506f1c51794677cb8d/packages/terra-functional-testing/src/reporters/file-output-reporter/index.js#L155